src/systems/diff_system.C

```diff
@@ -209,4 +192,174 @@ void DifferentiableSystem::release_linear_solver(LinearSolver<Number> *) const
+                                                      unsigned int dot_var_idx )
+{
+  // We're assuming that there could be a lot more variables than
+  // boundary conditions, so we search each of the boundary conditions
+  // for this variable rather than looping over boundary conditions
+  // in a separate loop and searching through all the variables.
+  const DirichletBoundaries * all_dbcs =
+    this->get_dof_map().get_dirichlet_boundaries();
+
+  if( all_dbcs )
+    {
+      // We need to cache the DBCs to be added so that we add them
+      // after looping over the existing DBCs. Otherwise, we're polluting
+      // the thing we're looping over.
+      std::vector<DirichletBoundary*> new_dbcs;
+
+      DirichletBoundaries::const_iterator dbc_it = all_dbcs->begin();
+      for( ; dbc_it != all_dbcs->end(); ++dbc_it )
+        {
+          libmesh_assert(*dbc_it);
+          DirichletBoundary & dbc = *(*dbc_it);
+
+          // Look for second order variable in the current
+          // DirichletBoundary object
+          std::vector<unsigned int>::const_iterator dbc_var_it =
+            std::find( dbc.variables.begin(), dbc.variables.end(), var_idx );
+
+          // If we found it, then we also need to add it's corresponding
+          // "dot" variable to a DirichletBoundary
+          std::vector<unsigned int> vars_to_add;
+          if( dbc_var_it != dbc.variables.end() )
+            vars_to_add.push_back(dot_var_idx);
+
+          if( !vars_to_add.empty() )
+            {
+              DirichletBoundary * new_dbc;
+
+              if( dbc.f )
+                {
+                  if( dbc.g )
+                    new_dbc = new DirichletBoundary(dbc.b,
+                                                    vars_to_add,
+                                                    dbc.f.get(),
+                                                    dbc.g.get());
+
+                  else
+                    new_dbc = new DirichletBoundary(dbc.b,
+                                                    vars_to_add,
+                                                    dbc.f.get());
+                }
+              else if( dbc.f_fem )
+                {
+                  if( dbc.g_fem )
+                    new_dbc = new DirichletBoundary(dbc.b,
+                                                    vars_to_add,
+                                                    *this,
+                                                    dbc.f_fem.get(),
+                                                    dbc.g_fem.get());
+
+                  else
+                    new_dbc = new DirichletBoundary(dbc.b,
+                                                    vars_to_add,
+                                                    *this,
+                                                    dbc.f_fem.get());
+                }
+              else
+                libmesh_error_msg("Could not find valid boundary function!");
```

Comment:
Didn't we decide that this was wrong?  The best we can do for now is add ZeroFunction DBCs for the dot vars, or scream-and-die if the primal var DBCs are FEM-dependent or time-dependent?